generators/python/src/fern_python/generators/core_utilities/core_utilities.py

```diff
@@ -259,14 +259,15 @@ def universal_field_validator(self, field_name: str, pre: bool = False) -> AST.F
         )
 
     def get_universal_root_model(self) -> AST.ClassReference:
-        if self._pydantic_compatibility == PydanticVersionCompatibility.Both:
+        if (
+            self._pydantic_compatibility == PydanticVersionCompatibility.Both
+            or self._pydantic_compatibility == PydanticVersionCompatibility.V2
+        ):
             return AST.ClassReference(
                 qualified_name_excluding_import=(),
                 import_=AST.ReferenceImport(
                     module=AST.Module.local(*self._module_path, "pydantic_utilities"), named_import="UniversalRootModel"
                 ),
             )
-        elif self._pydantic_compatibility == PydanticVersionCompatibility.V2:
-            return Pydantic(self._pydantic_compatibility).RootModel()
         else:  # V1 or V1_ON_V2
             return Pydantic(self._pydantic_compatibility).BaseModel()
```

Comment:
Discussed offline - the original implementation actually looks correct. My understanding of this is:
1. If you want to support both v1 and v2, we use our special `UniversalRootModel`.
2. If you only care about v2, we use `RootModel` (which superseded `BaseModel`).
3. If you only care about v1, we use `BaseModel`.

It sounds like the issue is actually that the `RootModel` inheritance is busted w.r.t. `mypy`, and we're using this setting to generate the IR. So instead of changing this implementation (and potentially breaking users), we should:

1. Try to move the Python IR package to `both`.
2. If (1) fails, fix the `RootModel` type hinting to make `mypy` happy.